README.md

```diff
@@ -19,6 +19,9 @@ Redlock works with Redis versions 6.0 or later.
 
 Redlock >= 2.0 only works with `RedisClient` instance.
 
+If you'd like to enable auto-reconnect attempts like in Redis 5,
+be sure to instantiate a RedisClient with `reconnect_attempts: 1`.
```

Comment:
aside, interesting logic inside where it defines `redis(&block)`  https://github.com/sidekiq/sidekiq/blob/v7.0.5/lib/sidekiq/config.rb#L118-L133

```ruby
    def redis
      raise ArgumentError, "requires a block" unless block_given?
      redis_pool.with do |conn|
        retryable = true
        begin
          yield conn
        rescue RedisClientAdapter::BaseError => ex
          # 2550 Failover can cause the server to become a replica, need
          # to disconnect and reopen the socket to get back to the primary.
          # 4495 Use the same logic if we have a "Not enough replicas" error from the primary
          # 4985 Use the same logic when a blocking command is force-unblocked
          # The same retry logic is also used in client.rb
          if retryable && ex.message =~ /READONLY|NOREPLICAS|UNBLOCKED/
            conn.close
            retryable = false
            retry
          end
          raise
        end
      end
    end
```


I'm doing some more comparisons since we got a flood of OpenSSL errors for about 10 minutes today which Heroku https://help.heroku.com/70UD7VSR/why-am-i-seeing-syscall-returned-5-errno-0-state-sslv3-tls-write-client-hello-when-trying-to-connect-to-heroku-redis reports as "because the maximum number of clients to Redis has been reached" which appears to be what happened
![image](https://user-images.githubusercontent.com/142914/219726118-6443d407-e237-412e-ac0a-0c5fefb46f40.png)

Nothing this PR would address, I don't think, but worth sharing in the upgrade changes discussion, I think.